### PR TITLE
Add CLTV (BIP65) support

### PR DESCRIPTION
**Pull request by eordano**
Mostly to be compatible with https://github.com/bitcoin/bitcoin/pull/6124

### CODE REVIEW

File: lib/script/interpreter.js [hunk 1]
```diff
@@ -184,6 +184,9 @@ Interpreter.false = new Buffer([]);
 
 Interpreter.MAX_SCRIPT_ELEMENT_SIZE = 520;
 
+Interpreter.LOCKTIME_THRESHOLD = 500000000;
+Interpreter.LOCKTIME_THRESHOLD_BN = new BN(500000000);
```
Comment (maraoz) [inline lib/script/interpreter.js]: I would do `new BN(Interpreter.LOCKTIME_THRESHOLD)`

File: lib/script/interpreter.js [hunk 2]
```diff
@@ -311,6 +317,52 @@ Interpreter.prototype.evaluate = function() {
...
 };
 
+/**
+ * Checks a locktime parameter with the transaction's locktime
+ *
+ * @param {BN} nLockTime the locktime read from the script
+ * @return {boolean} true if the transaction's locktime is less than or equal to
+ *                   the transaction's locktime
+ */
+Interpreter.prototype.checkLockTime = function(nLockTime) {
```
Comment (braydonf) [inline lib/script/interpreter.js]: We should likely reference `TransactionSignatureChecker::CheckLockTime` from Bitcoin Core in the comment here, and further information (such as comments?) can be found there.

**Comment by braydonf** [comment]
I think it would be good to add unit tests around `Interpreter.prototype.checkLockTime`

**Comment by eordano** [comment]
All right! Updated comments + test coverage.

File: lib/script/interpreter.js (hunk 2 above)
Comment (eordano) [inline lib/script/interpreter.js]: Updated (although for some reason this diff is not shown as outdated)

**Comment by braydonf** [comment]
LGTM

**Comment by braydonf** [comment]
The data for tx_invalid.json differs from the data in https://github.com/bitcoin/bitcoin/pull/6124/files Shouldn't these be identical? e.g. We should be able to copy tx_invalid.json directly.

**Comment by eordano** [comment]
Updated to match. Thanks for reviewing! I overlooked those tests by mistake

**Comment by braydonf** [comment]
Great! 

May be a good idea to submit to Bitcoin Core the test data for the last test "Failure due to empty stack before CHECKLOCKTIMEVERIFY in redeemScript".

**Pull request merged**